credentials/sts/sts_test.go

```diff
@@ -765,3 +775,184 @@ func (s) TestTokenInfoFromResponse(t *testing.T) {
 		})
 	}
 }
+
+func (s) TestTLS(t *testing.T) {
+
+	// start STS HTTPS server
+	stsServer := httptest.NewUnstartedServer(
+		http.HandlerFunc(
+			func(w http.ResponseWriter, r *http.Request) {
+				if r.URL.Path == "/token" {
+					// validate the inbound request
+					reqParams := &requestParameters{}
+					err := json.NewDecoder(r.Body).Decode(reqParams)
+					if err != nil {
+						fmt.Printf("Could Not parse STS Request application/json payload: (%+v)", err)
+						http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
+						return
+					}
+
+					// generate a static response
+					respParams := &responseParameters{
+						AccessToken:     accessTokenContents,
+						IssuedTokenType: issuedTokenType,
+						TokenType:       "Bearer",
+						ExpiresIn:       int64(60),
+					}
+					w.Header().Set("Content-Type", "application/json")
+					w.WriteHeader(http.StatusOK)
+					json.NewEncoder(w).Encode(respParams)
+				} else {
+					w.WriteHeader(http.StatusNotFound)
+				}
```

Comment:
Please handle the `else` part first up. That would un-indent the majority of the code.

```golang
if r.URL.Path != "/token" {
	w.WriteHeader(http.StatusNotFound)
	return
}
// Rest of the logic to validate the request and generate the response.
```